### DISCUSSION TITLE
LIS2DH tap interrupt never fires

### DISCUSSION THREAD

**Discussion by MaxMaeder** [zephyrproject-rtos/zephyr#71181] [Q&A]
Has anyone been able get the LIS2DH tap interrupt trigger working? I've modified the lis2dh sample to detect taps, and I can't get it to detect anything, as in my handler never fires.

It works fine for other triggers, for example `SENSOR_TRIG_DELTA`. I get no errors/return codes when registering the tap trigger, it just doesn't fire.

I'd really appreciate any help getting this working, as there isn't a ton of documentation around this.

Here is my modified code:
```conf
CONFIG_NEWLIB_LIBC=y
CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
CONFIG_I2C=y
CONFIG_SENSOR=y
CONFIG_LIS2DH=y
CONFIG_LIS2DH_TRIGGER_GLOBAL_THREAD=y
```
```c
#include <stdio.h>
#include <zephyr/kernel.h>
#include <zephyr/device.h>
#include <zephyr/drivers/sensor.h>

static void trigger_handler(const struct device *dev,
                            const struct sensor_trigger *trig)
{
    printf("TRIG\n");
}

int main(void)
{
    const struct device *sensor = DEVICE_DT_GET(DT_ALIAS(accel0));

    if (sensor == NULL || !device_is_ready(sensor))
    {
        printf("Could not get accel0 device\n");
        return -ENODEV;
    }

    struct sensor_trigger trig;
    int rc;

    trig.type = SENSOR_TRIG_TAP;
    rc = sensor_trigger_set(sensor, &trig, trigger_handler);
    if (rc != 0)
    {
        printf("Failed to set trigger: %d\n", rc);
        return rc;
    }

    printf("Waiting for triggers\n");
    while (true)
    {
        k_sleep(K_MSEC(2000));
    }

    return 0;
}
```

**Comment by MaxMaeder** [comment]
Bump, still haven't figured this out

**Reply by Ahtasham08** [reply]
Have you got the fix; I want to trigger two callbacks one for wakeup on movement other for double click detection?

**Comment by ttmut** [comment]
There seems to be two interrupt lines so maybe the other one is firing. Did you check voltage levels on these lines to see if interrupts are really firing? Also try setting `CONFIG_LOG=y` so that you may see error logs.
